front/src/Utils/DeviceUtils.ts

```diff
@@ -0,0 +1,68 @@
+import { detect } from './deviceDetection';
+const browser = detect();
+
+/**
+ * The DeviceUtils class is able to return information about the device used such as:
+ * the type of device, the screen, the OS, the web browser or even if it supports a specific JavaScript feature
+ * /!\ WARNING: Note that the above solution is not always reliable. The value of the userAgent can be easily changed.
+*/
+export class DeviceUtils {
+    private static message: string = "Your browser is compatible";
+
+    /**
+     * Add cases of incompatibilities here
+     */
+    public static isCompatible(): boolean {
+        // Not compatible with "iOS < 15" because it doesn't support "BigInt64Array"
+        if (browser?.os === "iOS" && browser?.version < 15) {
+            this.message = `Your browser is not compatible. Please update your ${browser.os} version (you have ${browser.version})`
+            return false;
+        }
+
+        // Not compatible with "edge < 79" because it doesn't support "BigInt64Array"
+        if (browser?.browser === "edge" && browser?.version < 79) {
+            this.message = `Your browser is not compatible. Please update your ${browser.browser} version (you have ${browser.version})`
+            return false;
+        }
+
+        // Not compatible with "firefox < 68" because it doesn't support "BigInt64Array"
+        if (browser?.browser === "firefox" && browser?.version < 68) {
+            this.message = `Your browser is not compatible. Please update your ${browser.browser} version (you have ${browser.version})`
+            return false;
+        }
+
+        // Not compatible with "chrome < 67" because it doesn't support "BigInt64Array"
+        if (browser?.browser === "chrome" && browser?.version < 67) {
+            this.message = `Your browser is not compatible. Please update your ${browser.browser} version (you have ${browser.version})`
+            return false;
+        }
+
+        // Not compatible with "android < 99" because it doesn't support "BigInt64Array"
+        if (browser?.browser === "android" && browser?.version < 99) {
+            this.message = `Your browser is not compatible. Please update your ${browser.browser} version (you have ${browser.version})`
+            return false;
+        }
+
+        return true;
```

Comment:
I'm not a huge fan of this code.

If someone comes with a very old browser (like internet explorer), he will not be told anything (!)

You should really instead check for the feature itself rather than the browser version.

Read this article to learn how to do it: https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Feature_detection

Now, I think it can be beneficial to have in the application a list of supported browsers with the minimum supported version.

Something like:

```
const supportedBrowsers = {
    "edge": 79,
    "chrome": 67
}
```
...

So the workflow should be:

1. apply feature detection
2. if missing feature:
    a.  detect browser
    b. if browser is part of the supported browsers, display a message with "upgrade to version ...."
    c. if browser is non part of the supported browsers, display a message with "your browser is not supported, install a supported browser (we recommend Chrome)"

Reply:
I think the step 2.b is too risky. We are betting on user agent, which is a data that can change, so I would rather try to detect the browser, if I don't find it I'll say that it is supported instead of unsupported.
Then if we find a new browser version that is not supported we simply add it to the list.

For all the rest I agree!

Reply:
2.c means: "if we detected a missing feature AND the browser is not supported", so we can definitely say "install a supported browser" and stop because we know it's going to fail anyway (the feature is missing)